spec/std/slice_spec.cr

```diff
@@ -104,25 +104,34 @@ describe "Slice" do
 
   it "does []? with start and count" do
     slice = Slice.new(4) { |i| i + 1 }
+
     slice1 = slice[1, 2]?
     slice1.should_not be_nil
     slice1 = slice1.not_nil!
     slice1.size.should eq(2)
+    slice1.to_unsafe.should eq(slice.to_unsafe + 1)
```

Comment:
Note: I added tests to verify that the subslice is pointing to the same memory as the original slice, and not returning a copy.